### ISSUE TITLE
AJV Example Validation Fails for Nested `$refs`

### ISSUE THREAD

**Issue by jd014** [Redocly/redocly-cli#965]
**Describe the bug**
When linting with rules `no-invalid-schema-examples` or `no-invalid-media-type-example` enabled and having nested `$ref` tags in properties, a `schema with key or id <pointer> already exists` error will be thrown. 

**To Reproduce**
1. Given the following files:
`test.yaml`
```
openapi: 3.1.0

info:
  title: Example API
  version: v1

servers:
  - description: Server
    url: https://domain.com

components:
  schemas:
    id:
      type: string
      format: uuid
    UserType1:
      type: object
      properties:
        id:
          $ref: '#/components/schemas/id'
          description: User ID
        anotherId:
          $ref: '#/components/schemas/id'
          description: Another ID
        name:
          type: string
      required:
        - id
        - name
    UserType2:
      type: object
      properties:
        id:
          $ref: '#/components/schemas/UserType1/properties/id'
        anotherId:
          $ref: '#/components/schemas/UserType1/properties/anotherId'
        age:
          type: number
      required:
        - id
        - age
      example:
        id: 9f5668cd-c3a3-4024-985c-2489715b2cb8
        age: 21

paths:
  /test:
    get:
      summary: Test
      responses:
        '200':
          description: OK
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/UserType2'

```
`.redocly.yaml`
```
apis:
  test:
    root: test.yaml
    extends:
      - minimal
    rules:
      no-invalid-schema-examples: error
      no-invalid-media-type-example: error
      info-license: off
      operation-operationId: off
      security-defined: off
      no-unused-components: off
```
2. Run `redocly lint test`
3. See error: `Example validation errored: schema with key or id "test.yaml#/components/schemas/id" already exists.`

**Expected behavior**
Nested `$ref` tags should not cause errors on validation.

**Redocly Version(s)**
1.0.0-beta.113

**Additional context**
The error seems to occur in `core/src/rules/ajv.ts` during `loadSchemaSync`. I was unable to find any documentation or references to `loadSchemaSync`. I assumed it was a custom function being used in Redocly's fork of ajv, but I still couldn't find any references in that repo either. Regardless, it seems to be creating a schema with a duplicate `$id` because it will resolve nested references to the same base schema. There needs to be some kind of cache to avoid loading the same schema multiple times: https://ajv.js.org/guide/managing-schemas.html#caching-schemas-in-your-code.

**Comment by IgorKarpiuk** [comment]
Hi @jd014. Thanks for reporting the bug. We will take a look.

**Comment by tatomyr** [comment]
I think the bug has been already fixed (works on 1.25.0 as expected). 

@jd014 please let me know if the issue is still there for you.
